### PR TITLE
[Feature]: Add phone number to the private personal details section

### PR DESCRIPTION
**Pull request by allgandalf**
### Details

Discussed in [this thread](https://expensify.slack.com/archives/C036QM0SLJK/p1725560507340979)

Users need a phone number to issue physical Expensify cards. We ask them for the phone number in the form to ship the Expensify card, but otherwise, this phone number is not editable from NewDot. This is a problem in case they want to edit it after adding it for the Expensify card.

### Fixed Issues

$ https://github.com/Expensify/App/issues/50701
PROPOSAL: https://expensify.slack.com/archives/C02NK2DQWUX/p1728885017172809?thread_ts=1728884992.974789&cid=C02NK2DQWUX

### Tests

1. Open App
2. Go to settings > profile
3. In private details, click on Phone Number

Verify that: you are able to enter phone number and styling is correct, and verify that error is displayed on wrong entry
- [x] Verify that no errors appear in the JS console

### Offline tests

1. Open App
2. Go to settings > profile
3. In private details, click on Phone Number

Verify that: you are able to enter phone number and in offline mode the number is displayed

### QA Steps

1. Open App
2. Go to settings > profile
3. In private details, click on Phone Number

Verify that: you are able to enter phone number and styling is correct, and verify that error is displayed on wrong entry

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/0427052a-21a5-4499-bb5d-bd6bdbcddfe2

</details>

<details>
<summary>Android: mWeb Chrome</summary>
Screenshot 2024-10-15 at 12 20 25 PM

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/fd681814-e252-415d-a0c4-caa3731cc466

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/28ce9a03-4533-4ae4-a743-824710225742

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/4ba2dbe9-7ab7-49d6-93b6-bbfddea8e30e

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/b48a9062-6277-4315-8ab8-8df9a6cd8a58

</details>

### CODE REVIEW

File: src/pages/settings/Profile/PersonalDetails/PhoneNumberPage.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    const {translate} = useLocalize();
+    const phoneNumber = privatePersonalDetails?.phoneNumber ?? '';
+    console.log('phoneNumber', phoneNumber);
+
+    const updatePhoneNumber = (values: PrivatePersonalDetails) => {
+        PersonalDetails.updatePhoneNumber(values?.phoneNumber ?? '');
+        Navigation.goBack();
+    };
+
+    const validate = useCallback(
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/pages/settings/Profile/PersonalDetails/PhoneNumberPage.tsx:36]: Lets use the one which is in the card shipping form and we can check out how to improve this later lets not block on it

File: src/libs/actions/PersonalDetails.ts [hunk 2]
```diff
@@ -157,6 +158,21 @@ function updateDateOfBirth({dob}: DateOfBirthForm) {
     Navigation.goBack();
 }
 
+function updatePhoneNumber(phoneNumber: string) {
+    const parameters: UpdatePhoneNumberParams = {phoneNumber};
+    API.write(WRITE_COMMANDS.UPDATE_PHONE_NUMBER, parameters, {
+        optimisticData: [
```
Comment (DylanDylann) [inline src/libs/actions/PersonalDetails.ts:164]: Should we follow the offline pattern B?

cc @mountiny

File: src/libs/actions/PersonalDetails.ts [hunk 3]
```diff
@@ -157,6 +158,21 @@ function updateDateOfBirth({dob}: DateOfBirthForm) {
     Navigation.goBack();
 }
 
+function updatePhoneNumber(phoneNumber: string) {
```
Comment (DylanDylann) [inline src/libs/actions/PersonalDetails.ts:161]: @allgandalf we need to handle error here

Comment (DylanDylann) [reply src/libs/actions/PersonalDetails.ts:161]: @mountiny Wdyt?

File: src/libs/actions/PersonalDetails.ts (hunk 2 above)
Comment (DylanDylann) [reply src/libs/actions/PersonalDetails.ts:164]: @allgandalf Bump

**Comment by dubielzyk-expensify** [comment]
I think the label should simply say `Phone number` instead of `What's your phone number`:
CleanShot 2024-10-16 at 10 16 20@2x

I also find the adding of the +1 after the fact a bit odd. Is this something we do in the rest of the app? cc @Expensify/design

File: src/libs/actions/PersonalDetails.ts (hunk 2 above)
Comment (mountiny) [reply src/libs/actions/PersonalDetails.ts:164]: Lets do the same we do for the other private personal details

Comment (mountiny) [reply src/libs/actions/PersonalDetails.ts:164]: How is the rbr handled there though? Like if there is an error when saving it we still have to show a rbr

File: src/libs/actions/PersonalDetails.ts (hunk 3 above)
Comment (mountiny) [reply src/libs/actions/PersonalDetails.ts:161]: We should add rbr in case of an error from api that will show error message under the input

**Comment by shawnborton** [comment]
Yup, agree with Jon's comments. 

Regarding the country code... yeah, that's an interesting one. Do we need to do that automatically or can we just rely on the user to enter one if needed?

File: src/libs/actions/PersonalDetails.ts (hunk 2 above)
Comment (mountiny) [reply src/libs/actions/PersonalDetails.ts:164]: so if there is an error on providing idk too long legal name, there is no feedback?

**Comment by dubielzyk-expensify** [comment]
> The country code is added by the BE automatically even when the user submits only number, so if we want to change that, we need to update the BE

I think the execution of this feels a bit inelegant, but I get why it's done this way. It kinda feels like there should be a country selector on that screen that'd defaulted to their location. But I also feel like all this is something worth solving later.

Thoughts @Expensify/design ?

**Comment by mountiny** [comment]
Yeah I think we could improve this with some dedicated country code selector, but I would say that is something for later.

@allgandalf @DylanDylann For FE validation, lets use the same as in the card shipping flow. 

For handling BE errors, show them using RBR dot on the row in the Private setting and then under the input with X to clear it

**Comment by shawnborton** [comment]
Yeah I agree with Jon that it doesn't feel the nicest, but I also see the point that if it ain't broke... we can revisit that particular part of the flow later.

**Comment by dannymcclain** [comment]
I'm in line with the comments above. A dedicated country picker would be nice on that screen, but I also agree that we don't need to do it right this second.

**Comment by DylanDylann** [comment]
Screenshot 2024-10-22 at 09 04 19

@allgandalf Let's add a padding-top on error message to ensure a space 8px as form error message

Screenshot 2024-10-22 at 09 05 11

**Comment by DylanDylann** [comment]
Let's add RBR on Profile tab and avatar tab when there is a error

Screenshot 2024-10-22 at 09 05 49

**Comment by DylanDylann** [comment]
@allgandalf 2 error messages are displayed at the same time

Screenshot 2024-10-22 at 09 09 44

**Review by allgandalf — COMMENTED** [review]
few comments

**Comment by DylanDylann** [comment]
https://github.com/user-attachments/assets/27faf248-2544-41a0-957a-ca194763a3f5

@allgandalf This is video to reproduce

**Comment by DylanDylann** [comment]
@allgandalf This make me confused

1. Enter 0387987987
2. Submit a request
3. The phone number is updated `+84387987987 `and an error message appears
4. Remove the error message
5. submit phone number again with no change, but we don't get error

https://github.com/user-attachments/assets/cf3abe53-334b-4b57-8dc2-760ffccfc69f

**Comment by DylanDylann** [comment]
> The phone number is not updated, we fallback to the previous phone number here, you can check in the video that you fallback to the previous phone number.

Kakaaa, wrong video. Wait a sec

**Comment by DylanDylann** [comment]
@dubielzyk-expensify Could you help to check [this comment](https://github.com/Expensify/App/pull/50773#issuecomment-2428503841) and please give your though on it? we have 2 error message displays at the same time.

File: src/pages/settings/Profile/PersonalDetails/PhoneNumberPage.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    const [privatePersonalDetails] = useOnyx(ONYXKEYS.PRIVATE_PERSONAL_DETAILS);
+    const [isLoadingApp] = useOnyx(ONYXKEYS.IS_LOADING_APP, {initialValue: true});
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const phoneNumber = privatePersonalDetails?.phoneNumber ?? '';
+
+    const validateLoginError = ErrorUtils.getEarliestErrorField(privatePersonalDetails, 'phoneNumber');
+    const currenPhoneNumber = privatePersonalDetails?.phoneNumber ?? '';
+
+    const updatePhoneNumber = (values: PrivatePersonalDetails) => {
```
Comment (DylanDylann) [inline src/pages/settings/Profile/PersonalDetails/PhoneNumberPage.tsx:35]: Please clear the error when submit phone number

**Comment by DylanDylann** [comment]
For [this](https://github.com/Expensify/App/pull/50773#issuecomment-2428503841), Let's use shouldHideFixErrorsAlert props. wdyt?

**Comment by DylanDylann** [comment]
@allgandalf Please prevent calling API if `the old value === the new value`

**Comment by dannymcclain** [comment]
>@dubielzyk-expensify Could you help to check https://github.com/Expensify/App/pull/50773#issuecomment-2428503841 and please give your though on it? we have 2 error message displays at the same time.

I don't think stacking the error message is terrible—I think the issue is that they're the _same_ error message. I think in this specific case it would be fine for the _form_ error to override the previous (dismissible) error message. cc @Expensify/design for thoughts.

**Comment by shawnborton** [comment]
Let me know if that makes sense, I realize after re-reading it that maybe I did not explain that super well :)

**Comment by dubielzyk-expensify** [comment]
I see what you mean and agree Shawn. I wonder if we error could just disappear `onChange` i.e. as soon as the user changes anything in the number field

**Comment by dubielzyk-expensify** [comment]
Yeah, I think we should show 1 error and just hide it when the users tries to rectify the error. But keen to hear if that's aligned with what @shawnborton meant

**Comment by DylanDylann** [comment]
@shawnborton  this is our expect, right?
> I see what you mean and agree Shawn. I wonder if we error could just disappear onChange i.e. as soon as the user changes anything in the number field

**Comment by shawnborton** [comment]
I was thinking as soon as the user submits, not types.

**Comment by DylanDylann** [comment]
@shawnborton Wdym?

> as the user submits

**Comment by DylanDylann** [comment]
https://github.com/user-attachments/assets/27faf248-2544-41a0-957a-ca194763a3f5

**Comment by DylanDylann** [comment]
@shawnborton You can point out the time on the above video

**Comment by shawnborton** [comment]
Ah okay, so then maybe we would need to do it onBlur at this moment here:
CleanShot 2024-10-23 at 12 02 28@2x

So not onChange, but onBlur. Could that work?

**Comment by shawnborton** [comment]
Nice, that feels pretty good to me. Curious what others think too!

**Comment by DylanDylann** [comment]
@shawnborton @dubielzyk-expensify  This is the current behavior in this flow

https://github.com/user-attachments/assets/2a8edf0a-8ebf-40c3-a470-e30589dbea69

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by dubielzyk-expensify** [comment]
The behavior in [this video](https://github.com/Expensify/App/pull/50773#issuecomment-2431598838) looks good to me. The last video posted doesn't seem to save the number which is a bit odd.

**Comment by shawnborton** [comment]
Which video particularly Jon? It looks like the two videos posted [here](https://github.com/Expensify/App/pull/50773#issuecomment-2434088835) are working as expected at least.

**Comment by dannymcclain** [comment]
OH I see what Jon is saying. In the [failure video here](https://github.com/Expensify/App/pull/50773#issuecomment-2434088835), we're not saving the "bad" phone number the user entered. Is that expected?

**Comment by shawnborton** [comment]
Ah, is that because it was considered a "bad" number from the backend? I have no idea how that works TBH.

**Comment by dubielzyk-expensify** [comment]
Yeah, I guess I was expecting it to save and show even though it throws an error. Happy to move forward as you've got control 👍

File: src/pages/settings/Profile/PersonalDetails/PhoneNumberPage.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        // Clear the error when the user tries to submit the form
+        if (validateLoginError) {
+            PersonalDetails.clearPhoneNumberError();
+        }
+
+        // Only call the API if the user has changed their phone number
+        if (phoneNumber !== values?.phoneNumber) {
```
Comment (DylanDylann) [inline src/pages/settings/Profile/PersonalDetails/PhoneNumberPage.tsx:42]: I think we should move thí check to above validateLoginError

**Comment by DylanDylann** [comment]
@allgandalf BUG: 2 error messages are display at the same time

https://github.com/user-attachments/assets/c7af9f9b-f6dd-4545-b973-cfe5d641a498

**Review by DylanDylann — APPROVED** [review]

**Review by DylanDylann — APPROVED** [review]

File: src/hooks/useIndicatorStatus.ts [hunk 6]
```diff
@@ -28,6 +28,7 @@ function useIndicatorStatus(): IndicatorStatusResult {
     const [userWallet] = useOnyx(ONYXKEYS.USER_WALLET);
     const [walletTerms] = useOnyx(ONYXKEYS.WALLET_TERMS);
     const [loginList] = useOnyx(ONYXKEYS.LOGIN_LIST);
+    const [privatePersoanlDetails] = useOnyx(ONYXKEYS.PRIVATE_PERSONAL_DETAILS);
```
Comment (mountiny) [inline src/hooks/useIndicatorStatus.ts:31]:
```suggestion
    const [privatePersonalDetails] = useOnyx(ONYXKEYS.PRIVATE_PERSONAL_DETAILS);
```

File: src/hooks/useIndicatorStatus.ts [hunk 7]
```diff
@@ -57,6 +58,7 @@ function useIndicatorStatus(): IndicatorStatusResult {
         [CONST.INDICATOR_STATUS.HAS_LOGIN_LIST_ERROR]: !!loginList && UserUtils.hasLoginListError(loginList),
         // Wallet term errors that are not caused by an IOU (we show the red brick indicator for those in the LHN instead)
         [CONST.INDICATOR_STATUS.HAS_WALLET_TERMS_ERRORS]: Object.keys(walletTerms?.errors ?? {}).length > 0 && !walletTerms?.chatReportID,
+        [CONST.INDICATOR_STATUS.HAS_PHONE_NUMBER_ERROR]: !!privatePersoanlDetails?.errorFields?.phoneNumber ?? undefined,
```
Comment (mountiny) [inline src/hooks/useIndicatorStatus.ts:61]:
```suggestion
        [CONST.INDICATOR_STATUS.HAS_PHONE_NUMBER_ERROR]: !!privatePersonalDetails?.errorFields?.phoneNumber ?? undefined,
```

File: src/pages/settings/InitialSettingsPage.tsx [hunk 8]
```diff
@@ -80,6 +80,7 @@ function InitialSettingsPage({currentUserPersonalDetails}: InitialSettingsPagePr
     const [walletTerms] = useOnyx(ONYXKEYS.WALLET_TERMS);
     const [loginList] = useOnyx(ONYXKEYS.LOGIN_LIST);
     const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY);
+    const [privatePersoanlDetails] = useOnyx(ONYXKEYS.PRIVATE_PERSONAL_DETAILS);
```
Comment (mountiny) [inline src/pages/settings/InitialSettingsPage.tsx:83]:
```suggestion
    const [privatePersonalDetails] = useOnyx(ONYXKEYS.PRIVATE_PERSONAL_DETAILS);
```

File: src/pages/settings/InitialSettingsPage.tsx [hunk 9]
```diff
@@ -126,7 +127,7 @@ function InitialSettingsPage({currentUserPersonalDetails}: InitialSettingsPagePr
      * @returns object with translationKey, style and items for the account section
      */
     const accountMenuItemsData: Menu = useMemo(() => {
-        const profileBrickRoadIndicator = UserUtils.getLoginListBrickRoadIndicator(loginList);
+        const profileBrickRoadIndicator = UserUtils.getProfilePageBrickRoadIndicator(loginList, privatePersoanlDetails);
```
Comment (mountiny) [inline src/pages/settings/InitialSettingsPage.tsx:130]:
```suggestion
        const profileBrickRoadIndicator = UserUtils.getProfilePageBrickRoadIndicator(loginList, privatePersonalDetails);
```

File: src/pages/settings/InitialSettingsPage.tsx [hunk 10]
```diff
@@ -161,7 +162,7 @@ function InitialSettingsPage({currentUserPersonalDetails}: InitialSettingsPagePr
         };
 
         return defaultMenu;
-    }, [loginList, fundList, styles.accountSettingsSectionContainer, bankAccountList, userWallet?.errors, walletTerms?.errors]);
+    }, [loginList, fundList, styles.accountSettingsSectionContainer, bankAccountList, userWallet?.errors, walletTerms?.errors, privatePersoanlDetails]);
```
Comment (mountiny) [inline src/pages/settings/InitialSettingsPage.tsx:165]:
```suggestion
    }, [loginList, fundList, styles.accountSettingsSectionContainer, bankAccountList, userWallet?.errors, walletTerms?.errors, privatePersonalDetails]);
```

File: src/libs/UserUtils.ts [hunk 11]
```diff
@@ -78,6 +78,23 @@ function getLoginListBrickRoadIndicator(loginList: OnyxEntry<LoginList>): LoginL
...
     }
+
+    return undefined;
+}
+
+/**
+ * Gets the appropriate brick road indicator status for the Profile section.
+ * Error status is higher priority, so we check for that first.
+ */
+function getProfilePageBrickRoadIndicator(loginList: OnyxEntry<LoginList>, privatePersoanlDetails: OnyxEntry<PrivatePersonalDetails>): LoginListIndicator {
```
Comment (mountiny) [inline src/libs/UserUtils.ts:89]:
```suggestion
function getProfilePageBrickRoadIndicator(loginList: OnyxEntry<LoginList>, privatePersonalDetails: OnyxEntry<PrivatePersonalDetails>): LoginListIndicator {
```

File: src/libs/UserUtils.ts [hunk 12]
```diff
@@ -78,6 +78,23 @@ function getLoginListBrickRoadIndicator(loginList: OnyxEntry<LoginList>): LoginL
...
+    const hasPhoneNumberError = !!privatePersoanlDetails?.errorFields?.phoneNumber;
```
Comment (mountiny) [inline src/libs/UserUtils.ts:90]:
```suggestion
    const hasPhoneNumberError = !!privatePersonalDetails?.errorFields?.phoneNumber;
```

**Review by mountiny — CHANGES_REQUESTED** [review]

**Review by mountiny — APPROVED** [review]
Thank you, easy-peasy 😂

**Pull request merged**